### PR TITLE
Fix MakeRegexpArray handling empty strings

### LINKED ISSUE #99616: kubectl proxy reject paths can't be empty
#### How to reproduce it (as minimally and precisely as possible):

```
$ kubectl proxy -p 8080 --reject-paths=""
```

```
$ kubectl exec -it podname -- sh
error: unable to upgrade connection: Forbidden
```

#### Environment:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by wzshiming**
#### What type of PR is this?

/kind bug
/sig cli

#### Which issue(s) this PR fixes:

Fixes #99616

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Comment by JornShen** [comment]
Adding some UT will be more perfect 🙃

**Comment by wzshiming** [comment]
> Adding some UT will be more perfect 🙃

Thanks, added

File: staging/src/k8s.io/kubectl/pkg/proxy/proxy_server.go [hunk 1]
```diff
@@ -61,6 +61,9 @@ type FilterServer struct {
 
 // MakeRegexpArray splits a comma separated list of regexps into an array of Regexp objects.
 func MakeRegexpArray(str string) ([]*regexp.Regexp, error) {
+	if str == "" {
```
Comment (dougsland) [inline staging/src/k8s.io/kubectl/pkg/proxy/proxy_server.go:64]: I would suggest:
```
if len(strings.TrimSpace(str)) == 0 {
```

This avoid also cases like:
```
 str := ""
 str := "    "
```

**Review by dougsland — COMMENTED** [review]
@wzshiming Looks good but I have nit comment

File: staging/src/k8s.io/kubectl/pkg/proxy/proxy_server.go (hunk 1 above)
Comment (soltysh) [reply staging/src/k8s.io/kubectl/pkg/proxy/proxy_server.go:64]: It would be nice, but not a hard requirement, imo.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**